test/MatrixFields/scalar_fieldmatrix.jl

```diff
@@ -0,0 +1,152 @@
+using Test
+using JET
+
+import ClimaCore: Geometry, Domains, Meshes, Spaces, Fields, MatrixFields
+import ClimaCore.Utilities: half
+import ClimaComms
+import ClimaCore.MatrixFields: @name
+ClimaComms.@import_required_backends
+include("matrix_field_test_utils.jl")
+
+@testset "get_field_first_index_offset" begin
+    struct Singleton{T}
+        x::T
+    end
+    struct TwoFields{T1, T2}
+        x::T1
+        y::T2
+    end
+    function test_get_field_first_index_offset(
+        name,
+        ::Type{T},
+        ::Type{S},
+        expected_offset,
+    ) where {T, S}
+        @test_all MatrixFields.get_field_first_index_offset(name, T, S) ==
+                  expected_offset
+    end
+    test_get_field_first_index_offset(
+        @name(x),
+        Float64,
```

Comment:
Can we set `FT` above from within `@testset "get_field_first_index_offset" begin` and use `FT` here?